samples/hybrid-connections/dotnet/portbridge/PortBridge/MultiplexConnectionInputPump.cs

```diff
@@ -66,9 +67,14 @@ public void DoneReadingPreamble(IAsyncResult readOutputAsyncResult)
                 int bytesRead = bufferRead.EndInvoke(readOutputAsyncResult);
                 if (bytesRead > 0)
                 {
-                    if (bytesRead == 1)
+                    while (bytesRead < preambleBuffer.Length)
                     {
-                        bytesRead += bufferRead(preambleBuffer, 1, preambleBuffer.Length - 1);
+                        int newBytesRead = this.bufferRead(preambleBuffer, bytesRead, preambleBuffer.Length - bytesRead);
+                        bytesRead += newBytesRead;
+                        if (newBytesRead == 0)
+                        {
+                            throw new ProtocolViolationException("Unexpected EOF while reading preamble");
+                        }
```

Comment:
this can go in a separate method?

Reply:
Done. Let me know if you have a better method name. I'm not totally sold on mine.

---
In reply to: [367089552](https://github.com/Azure/azure-relay/pull/50#discussion_r367089552) [](ancestors = 367089552)

Reply:
`assertNotEndOfStream` maybe? I was actually thinking the whole block could go in a method.

---
In reply to: [367151813](https://github.com/Azure/azure-relay/pull/50#discussion_r367151813) [](ancestors = 367151813,367089552)

Reply:
Ah, ok. Done in iteration 3.

---
In reply to: [367155549](https://github.com/Azure/azure-relay/pull/50#discussion_r367155549) [](ancestors = 367155549,367151813,367089552)